src/System.Linq.Parallel/tests/Combinatorial/SourcesAndOperators.cs

```diff
@@ -196,6 +214,32 @@ public static IEnumerable<object[]> UnaryFailingOperators()
             }
         }
 
+        public static IEnumerable<object[]> UnaryCancelingOperators()
+        {
+            // Token sources are supplied via delegate so they can be re-created as necessary.
+            // This means the funcs taking tokens and actions must be labeled.
+            LabeledOperation source = UnorderedRangeSources().First();
+            var src = Labeled.Label<Func<CancellationToken, Operation>>(source.ToString(), token => source.Append(WithCancellation(token)).Item);
+
+            foreach (var operation in new Labeled<Func<Action, Operation>>[] {
+                Labeled.Label<Func<Action, Operation>>("Distinct",cancel => (start, count, s) => s(start, count).Distinct(new CancelingEqualityComparer<int>(cancel))),
+                Labeled.Label<Func<Action, Operation>>("GroupBy-Comparer", cancel => (start, count, s) => s(start, count).GroupBy(x => x, new CancelingEqualityComparer<int>(cancel)).Select(g => g.Key)),
+                Labeled.Label<Func<Action, Operation>>("SelectMany", cancel => (start, count, s) => s(start, count).SelectMany(x => { cancel(); return new[] { x }; })),
+                Labeled.Label<Func<Action, Operation>>("SelectMany-Index", cancel => (start, count, s) => s(start, count).SelectMany((x, index) => { cancel(); return new[] { x }; })),
+                Labeled.Label<Func<Action, Operation>>("SelectMany-ResultSelector", cancel => (start, count, s) => s(start, count).SelectMany(x => new [] { x }, (group, elem) => { cancel(); return elem; })),
+                Labeled.Label<Func<Action, Operation>>("SelectMany-Index-ResultSelector", cancel => (start, count, s) => s(start, count).SelectMany((x, index) => new [] { x }, (group, elem) => { cancel(); return elem; })),
+                Labeled.Label<Func<Action, Operation>>("SkipWhile", cancel => (start, count, s) => s(start, count).SkipWhile(x => { cancel(); return true; })),
+                Labeled.Label<Func<Action, Operation>>("SkipWhile-Index", cancel => (start, count, s) => s(start, count).SkipWhile((x, index) => { cancel(); return true; })),
+                Labeled.Label<Func<Action, Operation>>("TakeWhile", cancel => (start, count, s) => s(start, count).TakeWhile(x => { cancel(); return true; })),
+                Labeled.Label<Func<Action, Operation>>("TakeWhile-Index", cancel => (start, count, s) => s(start, count).SkipWhile((x, index) => { cancel(); return true; })),
+                Labeled.Label<Func<Action, Operation>>("Where", cancel => (start, count, s) => s(start, count).Where(x => { cancel(); return true; })),
+                Labeled.Label<Func<Action, Operation>>("Where-Index", cancel => (start, count, s) => s(start, count).Where((x, index) => { cancel(); return true; })),
```

Comment:
Not easily, I don't think?  Besides the new comparators, they're calling a supplied action to enforce cancellation during operation run - and I specifically wanted it during the operation itself.  If I tried to do something like put a `Select` in front of these, the source might get repartitioned, and then the cancellation would happen on first `MoveNext`, instead of during the call.  
That, and the list is a subset.